# Data exploration TMA 02 Question 2b


I start by importing pandas to be able to work with dataframes and matplotlib to plot the data.

In [2]:
import pandas as pd

import matplotlib.pyplot as plt

Then I check the Deprivation_Index folder to see the different filenames of the documents.

In [3]:
!ls Data/

example2.csv


I decide to work with the examples.csv file, because there is a lot of interesting data in there about education and employment, the two areas I am interested in initally. To be able to analyse and compare the columns that contain the rankings, I removed the commas from the values using Excel, so that pandas sees them as integer. I saved the new csv-file as example2.csv.

I import the csv-file in a dataframe called df. To see the first five rows of the dataframe, I use head().

In [4]:
furl = 'Data/example2.csv'

df = pd.read_csv(furl)
df.head()

LSOAcode2011 LSOAname2011 LocalAuthorityDistrictcode2013  \
0    E01031349    Adur 001A                      E07000223   
1    E01031350    Adur 001B                      E07000223   
2    E01031351    Adur 001C                      E07000223   
3    E01031352    Adur 001D                      E07000223   
4    E01031370    Adur 001E                      E07000223   

  LocalAuthorityDistrictname2013  IndexofMultipleDeprivationRank  \
0                           Adur                           21352   
1                           Adur                            8864   
2                           Adur                           22143   
3                           Adur                           17252   
4                           Adur                           15643   

   IndexofMultipleDeprivationDecile  IncomeRank  IncomeDecile  EmploymentRank  \
0                                 7       18992             6           19305   
1                                 3        9233             3            7879   
2                                 7       24539             8           23389   
3                                 6       16087             5           13699   
4                                 5       17918             6           13322   

   EmploymentDecile  EducationSkillsandTrainingRank   \
0                 6                            13727   
1                 3                             6969   
2                 8                            10213   
3                 5                            10468   
4                 5                             7819   

   EducationSkillsandTrainingDecile  HealthDeprivationandDisabilityRank  \
0                                 5                               25876   
1                                 3                                6883   
2                                 4                               24693   
3                                 4                               19627   
4                                 3                               14893   

   HealthDeprivationandDisabilityDecile  CrimeRank  CrimeDecile   \
0                                     8      12817             4   
1                                     3      12781             4   
2                                     8       9112             3   
3                                     6      16127             5   
4                                     5      13723             5   

   BarrierstoHousingandServicesRank   BarrierstoHousingandServicesDecile   \
0                              28166                                    9   
1                              11399                                    4   
2                              24743                                    8   
3                              19559                                    6   
4                              22433                                    7   

   LivingEnvironmentRank   LivingEnvironmentDecile   
0                   18367                         6  
1                   16242                         5  
2                   22299                         7  
3                   22819                         7  
4                   18191                         6

To get a list of all the column names of the dataframe, I use list(). I can use these names later in my analysis.

In [6]:
list(df.columns)

['LSOAcode2011',
 'LSOAname2011',
 'LocalAuthorityDistrictcode2013',
 'LocalAuthorityDistrictname2013',
 'IndexofMultipleDeprivationRank',
 'IndexofMultipleDeprivationDecile',
 'IncomeRank',
 'IncomeDecile',
 'EmploymentRank',
 'EmploymentDecile',
 'EducationSkillsandTrainingRank ',
 'EducationSkillsandTrainingDecile',
 'HealthDeprivationandDisabilityRank',
 'HealthDeprivationandDisabilityDecile',
 'CrimeRank',
 'CrimeDecile ',
 'BarrierstoHousingandServicesRank ',
 'BarrierstoHousingandServicesDecile ',
 'LivingEnvironmentRank ',
 'LivingEnvironmentDecile ']

To get a first indication of the numbers in the dataframe, including the mean, minimum, and maximum value, I use describe().

In [7]:
df.describe()

IndexofMultipleDeprivationRank  IndexofMultipleDeprivationDecile  \
count                    32844.000000                      32844.000000   
mean                     16422.499391                          5.500122   
std                       9481.390618                          2.872325   
min                          1.000000                          1.000000   
25%                       8211.750000                          3.000000   
50%                      16422.500000                          5.500000   
75%                      24633.250000                          8.000000   
max                      32844.000000                         10.000000   

         IncomeRank  IncomeDecile  EmploymentRank  EmploymentDecile  \
count  32844.000000  32844.000000    32844.000000      32844.000000   
mean   16422.498660      5.500122    16422.498873          5.500122   
std     9481.390331      2.872325     9481.390246          2.872325   
min        1.000000      1.000000        1.000000          1.000000   
25%     8211.750000      3.000000     8211.750000          3.000000   
50%    16422.500000      5.500000    16422.500000          5.500000   
75%    24633.250000      8.000000    24633.250000          8.000000   
max    32844.000000     10.000000    32844.000000         10.000000   

       EducationSkillsandTrainingRank   EducationSkillsandTrainingDecile  \
count                     32844.000000                      32844.000000   
mean                      16422.499330                          5.500122   
std                        9481.390748                          2.872325   
min                           1.000000                          1.000000   
25%                        8211.750000                          3.000000   
50%                       16422.500000                          5.500000   
75%                       24633.250000                          8.000000   
max                       32844.000000                         10.000000   

       HealthDeprivationandDisabilityRank  \
count                        32844.000000   
mean                         16422.499726   
std                           9481.390467   
min                              1.000000   
25%                           8211.750000   
50%                          16422.500000   
75%                          24633.250000   
max                          32844.000000   

       HealthDeprivationandDisabilityDecile     CrimeRank  CrimeDecile   \
count                          32844.000000  32844.000000  32844.000000   
mean                               5.500122  16422.499635      5.500122   
std                                2.872325   9481.390606      2.872325   
min                                1.000000      1.000000      1.000000   
25%                                3.000000   8211.750000      3.000000   
50%                                5.500000  16422.500000      5.500000   
75%                                8.000000  24633.250000      8.000000   
max                               10.000000  32844.000000     10.000000   

       BarrierstoHousingandServicesRank   BarrierstoHousingandServicesDecile   \
count                       32844.000000                         32844.000000   
mean                        16422.499208                             5.500122   
std                          9481.390574                             2.872325   
min                             1.000000                             1.000000   
25%                          8211.750000                             3.000000   
50%                         16422.500000                             5.500000   
75%                         24633.250000                             8.000000   
max                         32844.000000                            10.000000   

       LivingEnvironmentRank   LivingEnvironmentDecile   
count            32844.000000              32844.000000  
mean             16422.499574                  5.500122  
std               94

These numbers look correct, as there are 32844 small areas in total and 10 deciles.

# Patterns in the data: national

I expect there will be a quite strong correlation between employment and education, so between the columns EmploymentRank and EducationSkillsandTrainingRank and between the columns EmploymentDecile and EducationSkillsandTrainingDecile. I expect this, apart from common sense, because the UK government mentioned in 2014 that the link between education and employment was strenghtening in the UK (source).

To see if this is also apparent from the data, I create the dataframe dfEmployment. It includes only the small areas in the worst three employment deciles. I use head() to see the first five rows of the dataframe.

In [8]:
dfEmployment = df[df.EmploymentDecile<4]

dfEmployment.head()

LSOAcode2011 LSOAname2011 LocalAuthorityDistrictcode2013  \
1     E01031350    Adur 001B                      E07000223   
16    E01031347    Adur 004A                      E07000223   
17    E01031348    Adur 004B                      E07000223   
18    E01031366    Adur 004C                      E07000223   
19    E01031371    Adur 004D                      E07000223   

   LocalAuthorityDistrictname2013  IndexofMultipleDeprivationRank  \
1                            Adur                            8864   
16                           Adur                            8184   
17                           Adur                            5675   
18                           Adur                            8423   
19                           Adur                            6060   

    IndexofMultipleDeprivationDecile  IncomeRank  IncomeDecile  \
1                                  3        9233             3   
16                                 3        6403             2   
17                                 2        5998             2   
18                                 3        9677             3   
19                                 2        6735             3   

    EmploymentRank  EmploymentDecile  EducationSkillsandTrainingRank   \
1             7879                 3                             6969   
16            8391                 3                             5156   
17            7003                 3                             3713   
18            8833                 3                             3833   
19            5896                 2                             2202   

    EducationSkillsandTrainingDecile  HealthDeprivationandDisabilityRank  \
1                                  3                                6883   
16                                 2                                9807   
17                                 2                                6763   
18                                 2                                7584   
19                                 1                                5996   

    HealthDeprivationandDisabilityDecile  CrimeRank  CrimeDecile   \
1                                      3      12781             4   
16                                     3      10509             4   
17                                     3       6131             2   
18                                     3       7379             3   
19                                     2      10491             4   

    BarrierstoHousingandServicesRank   BarrierstoHousingandServicesDecile   \
1                               11399                                    4   
16                              17114                                    6   
17                               9248                                    3   
18                              18378                                    6   
19                              12893                                    4   

    LivingEnvironmentRank   LivingEnvironmentDecile   
1                    16242                         5  
16                   11938                         4  
17                    9147                         3  
18                   14679                         5  
19                   17325                         6

I use describe() again to get a first indication of the numbers in the dataframe.

In [9]:
dfEmployment.describe()

IndexofMultipleDeprivationRank  IndexofMultipleDeprivationDecile  \
count                      9853.00000                       9853.000000   
mean                       5363.01157                          2.161981   
std                        3512.99666                          1.067717   
min                           1.00000                          1.000000   
25%                        2464.00000                          1.000000   
50%                        4959.00000                          2.000000   
75%                        7813.00000                          3.000000   
max                       19298.00000                          6.000000   

         IncomeRank  IncomeDecile  EmploymentRank  EmploymentDecile  \
count   9853.000000   9853.000000     9853.000000       9853.000000   
mean    5333.999188      2.152339     4926.999290          2.000101   
std     3502.738609      1.064998     2844.460361          0.816559   
min        1.000000      1.000000        1.000000          1.000000   
25%     2464.000000      1.000000     2464.000000          1.000000   
50%     4943.000000      2.000000     4927.000000          2.000000   
75%     7743.000000      3.000000     7390.000000          3.000000   
max    24256.000000      8.000000     9853.000000          3.000000   

       EducationSkillsandTrainingRank   EducationSkillsandTrainingDecile  \
count                      9853.000000                       9853.000000   
mean                       6628.079367                          2.549071   
std                        5316.113904                          1.613783   
min                           1.000000                          1.000000   
25%                        2531.000000                          1.000000   
50%                        5386.000000                          2.000000   
75%                        9419.000000                          3.000000   
max                       31669.000000                         10.000000   

       HealthDeprivationandDisabilityRank  \
count                         9853.000000   
mean                          5965.706790   
std                           4290.743832   
min                              1.000000   
25%                           2529.000000   
50%                           5225.000000   
75%                           8590.000000   
max                          29652.000000   

       HealthDeprivationandDisabilityDecile     CrimeRank  CrimeDecile   \
count                           9853.000000   9853.000000   9853.000000   
mean                               2.344565   9364.296052      3.370852   
std                                1.298958   7146.122367      2.165759   
min                                1.000000      1.000000      1.000000   
25%                                1.000000   3474.000000      2.000000   
50%                                2.000000   7820.000000      3.000000   
75%                                3.000000  13822.000000      5.000000   
max                               10.000000  32309.000000     10.000000   

       BarrierstoHousingandServicesRank   BarrierstoHousingandServicesDecile   \
count                        9853.000000                          9853.000000   
mean                        15281.146859                             5.157110   
std                          9230.760837                             2.806176   
min                             3.000000                             1.000000   
25%                          7200.000000                             3.000000   
50%                         15050.000000                             5.000000   
75%                         23011.000000                             8.000000   
max                         32833.000000                            10.000000   

       LivingEnvironmentRank   LivingEnvironmentDecile   
count             9853.000000               9853.000000  
mean             13594.003045                  4.647011  
std               89

The EducationSkillsandTrainingDecile mean of all the small areas that belong to the three worst employment deciles is 2.549071. This is only 0.55 higher than the EmploymentDecile mean and therefore a first indication that both columns are correlated.

To visualise the correlation, I create a bar chart using .plot(kind='bar')

The chart shows that most small areas that belong to the three most Employment deprived deciles also belong to the three most Education deprived deciles.

To confirm the correlation between Education and Employment, we can also look at the rankings. I therefore create a scatter graph to compare EducationSkillsandTrainingRank and EmploymentRank. I expect that an area with a high rank in Education also has a high rank in Employment.

The chart confirms that there is a correlation, although it is a bit widespread. To test how strong the correlation is, I can perform a statistical test. A suitable test for this correlation is Pearson’s R2. I therefore import the test in Python so I can use it.

In [14]:
import scipy
from scipy.stats.stats import pearsonr

Then I perform the test with the columns EducationSkillsandTrainingRank and EmploymnetRank.

In [15]:
x = df['EducationSkillsandTrainingRank ']
y = df['EmploymentRank']
scipy.stats.pearsonr(x, y)

(0.825013066391737, 0.0)

The relationship between two variables is generally considered strong when their r value is larger than 0.7 (source), so we can call this a strong correlation.